services/src/main/java/org/jboss/windup/web/services/messaging/WindupExecutionTask.java

```diff
@@ -122,6 +134,17 @@ public void run()
 
             Map<String, Object> otherOptions = getOtherOptions(analysisContext);
 
+            RulePhaseFinder phaseFinder = furnace.getAddonRegistry().getServices(RulePhaseFinder.class).get();
```

Comment:
I know that I have already said some of this, but I am not very fond of this approach. There are actually a few problems here:

- RuleProviderWithDependenciesPredicate is generally designed for use in tests, where you for certain that you want to only execute a particular subset of rules up to a certain point. This task isn't really about skipping all rules after a certain point, it is about skipping the report rendering phases specifically. Finalization related tasks should still execute.
- Windup core APIs should not be exposed to the services war unless necessary. In general, windup-web-war should only know about things in the web support addon. I realize that we might break this when absolutely necessary, but this approach, IMO is mixing concerns.
- I personally find the code below pretty difficult to read. It seems like a RulePhaseFilter would be simpler to read than this.